python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+                tol=self.config.tolerance,
+                maxSepInSigma=self.config.maxSeparation,
+                fitBgGradient=self.config.fitBgGradient,
+                separateNegParams=self.config.fitSeparateNegParams,
+                verbose=self.config.verbose, display=False)
+        except pexExcept.LengthError:
+            self.fail(measRecord, measBase.MeasurementError('edge failure', self.FAILURE_EDGE))
+        except Exception:
+            self.fail(measRecord, measBase.MeasurementError('dipole fit failure', self.FAILURE_FIT))
+
+        if result is None:
+            return result
+
+        self.log.log(self.log.DEBUG, "Dipole fit result: %s" % str(result))
+
+        if result.psfFitPosFlux <= 1.:   # usually around 0.1 -- the minimum flux allowed -- i.e. bad fit.
+            self.fail(measRecord, measBase.MeasurementError('dipole fit failure', self.FAILURE_FIT))
+
+        # add chi2, coord/flux uncertainties (TBD), dipole classification
+
+        # Add the relevant values to the measRecord
+        measRecord[self.posFluxKey] = result.psfFitPosFlux
+        measRecord[self.posFluxSigmaKey] = result.psfFitSignaltoNoise   # to be changed to actual sigma!
+        measRecord[self.posCentroidKeyX] = result.psfFitPosCentroidX
+        measRecord[self.posCentroidKeyY] = result.psfFitPosCentroidY
+
+        measRecord[self.negFluxKey] = result.psfFitNegFlux
+        measRecord[self.negFluxSigmaKey] = result.psfFitSignaltoNoise   # to be changed to actual sigma!
+        measRecord[self.negCentroidKeyX] = result.psfFitNegCentroidX
+        measRecord[self.negCentroidKeyY] = result.psfFitNegCentroidY
+
+        # Dia source flux: average of pos+neg
+        measRecord[self.fluxKey] = (abs(result.psfFitPosFlux) + abs(result.psfFitNegFlux))/2.
+        measRecord[self.orientationKey] = result.psfFitOrientation
+        measRecord[self.separationKey] = np.sqrt((result.psfFitPosCentroidX - result.psfFitNegCentroidX)**2. +
+                                                 (result.psfFitPosCentroidY - result.psfFitNegCentroidY)**2.)
+        measRecord[self.centroidKeyX] = (result.psfFitPosCentroidX + result.psfFitNegCentroidX)/2.
+        measRecord[self.centroidKeyY] = (result.psfFitPosCentroidY + result.psfFitNegCentroidY)/2.
+
+        measRecord[self.signalToNoiseKey] = result.psfFitSignaltoNoise
+        measRecord[self.chi2dofKey] = result.psfFitRedChi2
+
+        self.doClassify(measRecord, result)
+
+    def doClassify(self, measRecord, result):
+        """Determine if source is classified as dipole (similar to
+        orig. dipole classification task).
+        """
+
+        # First, does the total signal-to-noise surpass the minSn?
+        passesSn = measRecord[self.signalToNoiseKey] > self.config.minSn
+
+        # Second, are the pos/neg fluxes less than 1.0 and no more than 0.65 of the total flux?
+        # By default this will never happen since posFlux = negFlux.
```

Comment:
This comment does not seem to match the code. Also, hard-coded values worry me; shouldn't they be config params?